File: src/Java/Images/Actor2D.java
import vtk.vtkNamedColors;
import vtk.vtkNativeLibrary;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;
import vtk.vtkPoints;
import vtk.vtkVertexGlyphFilter;
import vtk.vtkPolyData;
import vtk.vtkPolyDataMapper2D;
import vtk.vtkActor2D;

public class Actor2D 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String s[]) 
  {

    vtkNamedColors colors = new vtkNamedColors();
   
    //For Actor Color
    double actorColor[] = new double[4];
    //Renderer Background Color
    double Bgcolor[] = new double[4];

    colors.GetColor("Lime", actorColor);
    colors.GetColor("DarkSlateGray", Bgcolor);

    vtkPoints Points = new vtkPoints();
    Points.InsertNextPoint(10, 10, 0);
    Points.InsertNextPoint(100, 100, 0);
    Points.InsertNextPoint(200, 200, 0);

    vtkPolyData PolyData = new vtkPolyData();
    PolyData.SetPoints(Points);

    vtkVertexGlyphFilter GlyphFilter = new vtkVertexGlyphFilter();
    GlyphFilter.SetInputData(PolyData);
    GlyphFilter.Update();

    vtkPolyDataMapper2D Mapper = new vtkPolyDataMapper2D();
    Mapper.SetInputConnection(GlyphFilter.GetOutputPort());
    Mapper.Update();

    vtkActor2D Actor = new vtkActor2D();
    Actor.SetMapper(Mapper);
    Actor.GetProperty().SetColor(actorColor);
    Actor.GetProperty().SetPointSize(8);

    // Create the renderer, render window and Interactor.
    vtkRenderer ren = new vtkRenderer();
    vtkRenderWindow renWin = new vtkRenderWindow();
    renWin.AddRenderer(ren);
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);

    // Add the actor to the scene
    ren.AddActor(Actor);
    ren.SetBackground(Bgcolor);

    renWin.SetSize(300, 300);
    renWin.Render();

    iren.Initialize();
    iren.Start();
  }
}

File: src/Python/Images/Actor2D.py
#!/usr/bin/env python

import vtk

def main():
    
    colors = vtk.vtkNamedColors()
    points = vtk.vtkPoints()
    points.InsertNextPoint(10,10,0)
    points.InsertNextPoint(100,100,0)
    points.InsertNextPoint(200,200,0)
    
    polydata = vtk.vtkPolyData()
    polydata.SetPoints(points)
    
    glyphFilter = vtk.vtkVertexGlyphFilter()
    glyphFilter.SetInputData(polydata)
    glyphFilter.Update()
    
    mapper = vtk.vtkPolyDataMapper2D()
    mapper.SetInputConnection(glyphFilter.GetOutputPort())
    mapper.Update()
    
    actor = vtk.vtkActor2D()
    actor.SetMapper(mapper)
    
    # Create a renderer, render window, and interactor
    renderer = vtk.vtkRenderer()
    renderWindow = vtk.vtkRenderWindow()
    renderWindow.AddRenderer(renderer)
    renderWindowInteractor = vtk.vtkRenderWindowInteractor()
    renderWindowInteractor.SetRenderWindow(renderWindow)
    
    # Add the actor to the scene
    renderer.AddActor(actor)
    renderWindow.SetSize(300,300)
    actor.GetProperty().SetColor(colors.GetColor3d("green")) 
    actor.GetProperty().SetPointSize(6)
    
    # Render and interact
    renderWindow.Render()
    w2if = vtk.vtkWindowToImageFilter()
    w2if.SetInput(renderWindow)
    w2if.Update()
    
    writer = vtk.vtkPNGWriter()
    writer.SetFileName("TestActor2D.png")
    writer.SetInputConnection(w2if.GetOutputPort())
    writer.Write()
    renderWindowInteractor.Start()

if __name__ == '__main__':
    main()
